### ISSUE TITLE
agStressPerTiter & srStressPerTiter ignore missing values

### ISSUE THREAD

**Issue by drserajames** [acorg/Racmacs#78]
I noticed that the agStressPerTiter was different to the average stress in the view window. I think these functions aren't calculating the stress per titre because they don't exclude missing ("*") titres.

**Comment by shwilks** [comment]
Do you have an example map where you're seeing this sort of thing Sarah?

**Comment by drserajames** [comment]
```
set.seed(850909)

dat <- matrix(10*2^round(10*runif(100)), ncol=10)
dat[4,3:5] <- "*"

map <- make.acmap(dat)

cbind(agStress(map), agStressPerTiter(map))[4,]
```
> cbind(agStress(map), agStressPerTiter(map))[4,]
[1] 11.693437  1.169344

Stress per titre from map is 1.67
<img width="626" alt="Screenshot 2022-03-01 at 11 33 17" src="https://user-images.githubusercontent.com/18366410/156161970-ef1da452-c3af-4ee2-84cd-82930ce72b96.png">

**Comment by shwilks** [comment]
_perfect_ thank you, I'll take a look what's going on based on your pointers

**Comment by shwilks** [comment]
looks like this perhaps actually stems from whether to exclude nds (<10s) or not when calculating the stress per titer.  In the viewer the number shown is when nds are not included (since stress is often 0 for those and it skews things) but the default for agStressPerTiter is not to exclude the <10s when taking the mean.  Due to other errors in the code I actually got NA when I did `cbind(agStress(map), agStressPerTiter(map, exclude_nd = T))[4,]` but with some fixes it now returns `11.693437  1.670491` so matches the viewer.  After the fix do you think we should change the default for `agStressPerTiter()` to `exclude_nd = T`.  Mostly this is the more informative number (hence it being the default in the viewer) so perhaps that would make sense?

**Comment by drserajames** [comment]
In most cases including <10s will decrease the stress per titre, so I would favour excluding them. I've gone over a few variants in my head (maximum of including & excluding nds, only including nds if they contribute to stress, only including nds if they increase the stress per titre). Perhaps the first one is worth sacrificing simplicity? It would alert the user to the case where there was a nd titre with a close distance (e.g. all other titres place it close to that serum). Or you could make the result a matrix, and have one column for each? Then the user has to choose what is right for them.

What is `agStressPerTiter(map)[4]` after the fix?

**Comment by shwilks** [comment]
The base problem is now fixed in v1.1.28, although I haven't changed the default setting for exclude_nd yet.  I like your solution of having it return e.g. a matrix so that you have to choose, maybe we should change it to that

**Comment by shwilks** [comment]
In v1.1.31 these functions now return a matrix with columns nd_included and nd_excluded. Will close this issue unless there are other things we might want to change
